### PR TITLE
fix: Next, Retry button issues and double blick stacking issue when pressing the button (FM-328)

### PR DESCRIPTION
**Pull request by woodsjohn-codev**
# Changes
- fix level end buttons and button click audio

# How to test
- Open the start screen of FTM.
- Tap or click any part of the start screen to proceed to the next screen
- Select any level
- Play the game until the end
- In level end scene
- Press any button
- Make sure button click audio only play once
- Make sure it proceeds to next level
- Make sure Retry button is working
- Make sure Map button is working

Ref: [FM-328](https://curiouslearning.atlassian.net/browse/FM-328)

[FM-328]: https://curiouslearning.atlassian.net/browse/FM-328?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by bernhardccodev — APPROVED** [review]
approved

File: src/scenes/levelend-scene/levelend-scene.ts [hunk 1]
```diff
@@ -27,28 +25,23 @@ export class LevelEndScene {
...
     this.data = data;
     this.audioPlayer = new AudioPlayer();
     this.canvasElement = document.getElementById("rivecanvas") as HTMLCanvasElement;
     this.starCount = starCount;
     this.currentLevel = currentLevel;
-    this.isLastLevel = this.currentLevel ===
-    this.data.levels[this.data.levels.length - 1].levelMeta.levelNumber;
+    this.isLastLevel =
+      this.currentLevel ===
+      this.data.levels[this.data.levels.length - 1].levelMeta.levelNumber;
```
Comment (miguelccodev) [inline src/scenes/levelend-scene/levelend-scene.ts:44]: This should be computed from `gameStateService`

Comment (woodsjohn-codev) [reply src/scenes/levelend-scene/levelend-scene.ts]: done here. please check @miguelccodev

File: src/gameStateService/game-state-service.ts [hunk 2]
```diff
@@ -171,6 +184,17 @@ export class GameStateService extends PubSub {
     getStonePositions() {
         return createStonePositionsDAO(this);
     }
+    
+    levelEndSceneData({levelEndData, data}) {
+        this.levelEndData = {...levelEndData};
```
Comment (miguelccodev) [inline src/gameStateService/game-state-service.ts:189]: Is there a need to store scene data in the game state?

Check if:
- it is shared across different components
- only used to compute state

File: src/gameStateService/game-state-service.ts [hunk 3]
```diff
@@ -171,6 +184,17 @@ export class GameStateService extends PubSub {
...
+        this.data = data;
+        this.isLastLevel =
+        levelEndData.currentLevel === data.levels[data.levels.length - 1].levelMeta.levelNumber;
+    }
+
+    getLevelEndSceneData() {
```
Comment (miguelccodev) [inline src/gameStateService/game-state-service.ts:195]: We should share generic data in the game state, unless it is common and shared across different components.

Does this need to sit here or can it be local in the end scene component?

File: src/gameStateService/data-access-objects.ts [hunk 4]
```diff
@@ -101,4 +103,13 @@ export const createStonePositionsDAO = ({
 	return randomizedStonePositions;
 }
 
+export const createLevelEndDataDAO = (gameState) => {
+  return {
+		starCount: gameState.levelEndData.starCount,
+		currentLevel: gameState.levelEndData.currentLevel,
+		isTimerEnded: gameState.levelEndData.isTimerEnded,
+		data: gameState.data
```
Comment (miguelccodev) [inline src/gameStateService/data-access-objects.ts:111]: These are common data to all scenes.

Can you add a todo to organize this later?

File: src/gameStateService/data-access-objects.ts [hunk 5]
```diff
@@ -101,4 +103,13 @@ export const createStonePositionsDAO = ({
 	return randomizedStonePositions;
 }
 
+export const createLevelEndDataDAO = (gameState) => {
+  return {
+		starCount: gameState.levelEndData.starCount,
```
Comment (miguelccodev) [inline src/gameStateService/data-access-objects.ts:108]: Does this need to sit here as a globally shared data or can it be local on the level end scene?

File: src/gameStateService/game-state-service.ts (hunk 3 above)
Comment (woodsjohn-codev) [reply src/gameStateService/game-state-service.ts:195]: To me, yes, this should be part of the game state. That way, if we need any of the data in the LevelEndScene, we can easily extract it.

**Review by miguelccodev — APPROVED** [review]

**Pull request merged**